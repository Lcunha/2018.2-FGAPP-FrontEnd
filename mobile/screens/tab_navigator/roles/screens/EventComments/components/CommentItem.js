import {
	Body,
	Button,
	Icon,
	Card,
	CardItem,
	Text,
	Thumbnail,
	Left,
	Right,
	Row,
	Item,
	ActionSheet,
	Root
} from "native-base";
import React, { Component } from "react";
import {
	StyleSheet,
	Image,
	View,
	TouchableOpacity,
	ScrollView
} from "react-native";
import Divider from "../../EventProfile/components/Divider";
import LikeButton from "./LikeButton";

var BUTTONS = [
	{ text: "Responder", icon: "quote", iconColor: "#2c8ef4" },
	{ text: "Editar", icon: "hammer", iconColor: "#f42ced" },
	{ text: "Delete", icon: "trash", iconColor: "#fa213b" },
	{ text: "Cancel", icon: "close", iconColor: "#25de5b" }
];
var DESTRUCTIVE_INDEX = 2;
var CANCEL_INDEX = 3;

class CommentItem extends Component {
	constructor(props) {
		super(props);
		this.state = {};
	}

	render() {
		return (
			<Root>
				<View>
					<Card transparent>
						<CardItem>
							<View
								style={{
									alignSelf: "flex-start",
									alignContent: "flex-start",
									paddingTop: 15
								}}
							>
								<Icon name="person" />
							</View>
							<View style={{ width: "100%" }}>
								<View flex={1}>
									<View
										style={{
											flexDirection: "row",
											alignContent: "space-between",
											width: "80%",
											justifyContent: "space-between"
										}}
									>
										<Text
											style={{
												fontWeight: "bold",
												textAlignVertical: "center"
											}}
										>
											{this.props.author}
										</Text>
										<Button
											style={{}}
											transparent
											onPress={() =>
												ActionSheet.show(
													{
														options: BUTTONS,
														cancelButtonIndex: CANCEL_INDEX,
														destructiveButtonIndex: DESTRUCTIVE_INDEX,
														title: "ComentÃ¡rio de " + this.props.author
													},
													buttonIndex => {
														this.setState({ clicked: BUTTONS[buttonIndex] });
													}
												)
											}
										>
											<Icon name="more" />
										</Button>
									</View>

									<Text>{this.props.text}</Text>

									<Text style={{ color: "grey", fontSize: 12 }}>
										Postado em: {this.props.postDate}
									</Text>
									{this.props.modifyDate != null && (
										<Text style={{ color: "grey", fontSize: 12 }}>
											Modificado em: {this.props.modifyDate}
										</Text>
									)}
								</View>
								<LikeButton />
							</View>
						</CardItem>
					</Card>

					<Divider size="80%" />
				</View>
			</Root>
		);
	}
}

export default CommentItem;
