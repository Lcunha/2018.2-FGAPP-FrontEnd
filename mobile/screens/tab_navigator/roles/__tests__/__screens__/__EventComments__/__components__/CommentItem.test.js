import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';
import CommentItem from '../../../../screens/EventComments/components/CommentItem'

Enzyme.configure({adapter: new Adapter()});

it('renders correctly', () => {
  const tree = renderer.create(<CommentItem/>).toJSON();
  expect(tree).toMatchSnapshot();
});