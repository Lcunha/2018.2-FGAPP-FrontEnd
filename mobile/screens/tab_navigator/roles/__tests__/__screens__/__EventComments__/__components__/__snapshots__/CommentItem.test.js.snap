// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
  virtual={undefined}
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "width": "80%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
      />
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "fontWeight": "bold",
              "textAlignVertical": "center",
            },
          ]
        }
        uppercase={false}
        virtual={undefined}
      />
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": null,
            "borderBottomWidth": null,
            "borderColor": "#007aff",
            "borderLeftWidth": null,
            "borderRadius": 5,
            "borderRightWidth": null,
            "borderTopWidth": null,
            "borderWidth": undefined,
            "elevation": 0,
            "flexDirection": "row",
            "height": 45,
            "justifyContent": "space-between",
            "opacity": 1,
            "paddingBottom": 6,
            "paddingTop": 6,
            "shadowColor": null,
            "shadowOffset": null,
            "shadowOpacity": null,
            "shadowRadius": null,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        />
      </View>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
      uppercase={false}
      virtual={undefined}
    />
  </View>
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={false}
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hasTVPreferredFocus={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.4)",
          "flex": 1,
          "justifyContent": "flex-end",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "elevation": 4,
            "height": NaN,
            "maxHeight": 667,
            "minHeight": 56,
            "opacity": 1,
            "padding": 15,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <RCTScrollView
          data={Array []}
          disableVirtualization={false}
          getItem={[Function]}
          getItemCount={[Function]}
          horizontal={false}
          initialNumToRender={10}
          invertStickyHeaders={undefined}
          keyExtractor={[Function]}
          maxToRenderPerBatch={10}
          numColumns={1}
          onContentSizeChange={[Function]}
          onEndReachedThreshold={2}
          onLayout={[Function]}
          onMomentumScrollEnd={[Function]}
          onScroll={[Function]}
          onScrollBeginDrag={[Function]}
          onScrollEndDrag={[Function]}
          renderItem={[Function]}
          scrollEventThrottle={50}
          stickyHeaderIndices={Array []}
          style={
            Object {
              "marginHorizontal": -15,
              "marginTop": 0,
            }
          }
          updateCellsBatchingPeriod={50}
          viewabilityConfigCallbackPairs={Array []}
          windowSize={21}
        >
          <View />
        </RCTScrollView>
      </View>
    </View>
  </Modal>
</View>
`;
